### PR TITLE
Use data from Studio for primary navigation

### PR DESCRIPTION
**Pull request by j0nas**
# Use data from Studio for primary navigation

## Intent

Replaces the hardcoded items in _Nav.tsx_  with items fetched from Studio.
Solves [this](https://app.shortcut.com/sanity-io/story/16033/use-data-from-studio-primary-navigation-as-header-nav) Shortcut Story.

## Testing this PR

Step-by-step instructions which reviewers can use to verify that this PR
achieves its aim. Specify any prerequisites and assumptions. Be explicit.

1. Open the [Preview Deploy link](https://structured-content-2022-web-git-use-data-from-studio-pri-7f9396.sanity.build/) for this branch
2. Scroll down past the NavBlock to the point where the primary nav appears
3. Verify that the rendered items correspond with the data available in Studio
4. (optional) Modify the Primary navigation items and verify upon rebuilding the site that the updated links render in accordance with the Studio the changes you made

### CODE REVIEW

File: web/pages/speakers/[slug].tsx [hunk 1]
```diff
@@ -1,31 +1,33 @@
...
+import type { Slug } from '../../types/Slug';
+import type { Session } from '../../types/Session';
+import { mainEventId } from '../../util/constants';
+import { sessionStart } from '../../util/session';
+import { PRIMARY_NAV, SPEAKER } from '../../util/queries';
 import styles from '../app.module.css';
 import speakerStyles from './speakers.module.css';
-import { sessionStart } from '../../util/session';
-import { Session } from '../../types/Session';
+import { PrimaryNavItem } from '../../types/PrimaryNavItem';
```
Comment (oyvind-stenhaug) [inline web/pages/speakers/[slug].tsx:24]: Nit: Can you share the convention you've adopted for ordering imports, so we can cut down on the diffs related to this going forward? (Assuming that eslint isn't configured to check it, which it doesn't seem to be.)

And is it a mistake that this type import isn't grouped along with the others further up (Person, Slug etc)?

**Review by oyvind-stenhaug — CHANGES_REQUESTED** [review]
The "this" link in the PR description seems to be incorrect.

As indicated in the inline comment on the query, the `slug` also can't just be used directly as `href` in the `MenuItem` component as written. Seems to me that the cleanest way to handle it would be to pass `internal` through `getEntityPath`, just like e.g. `SimpleCallToAction` does it. That way the conversion from slug to URL is centralized in one location.

To be very concrete: As written, on https://structured-content-2022-web-git-use-data-from-studio-pri-7f9396.sanity.build/speakers/carrie-hane all the internal Nav links are broken because they resolve to https://structured-content-2022-web-git-use-data-from-studio-pri-7f9396.sanity.build/speakers/program etc.

File: web/pages/speakers/[slug].tsx (hunk 1 above)
Comment (j0nas) [reply web/pages/speakers/[slug].tsx:24]: I decided to address this properly in #137.

File: web/util/queries.ts [hunk 2]
```diff
@@ -105,6 +105,14 @@ const SPONSORSHIP = `
   type,
 `;
 
+const PRIMARY_NAV = `
+  *[_type == "navigation" && slug.current == "primary-nav"][0].items[] {
+    label,
+    "href": coalesce(target.internal->.slug.current, target.external),
```
[the remark this answers is not in the document]
Comment (j0nas) [reply web/util/queries.ts]: Excellent feedback! This is how I should have done it in the first place: 0a89c08

**Comment by oyvind-stenhaug** [comment]
@j0nas There are some merge conflicts here that need resolving. Other than that I'm ready to approve

**Review by oyvind-stenhaug — APPROVED** [review]

**Pull request merged**